src/Stripe.net/Infrastructure/JsonConverters/ExternalAccountConverter.cs

```diff
@@ -24,21 +22,17 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
         {
             var incoming = JObject.Load(reader);
 
-            var externalAccount = new ExternalAccount
-            {
-                Id = incoming.SelectToken("id").ToString()
-            };
+            var externalAccount = default(IExternalAccount);
```

Comment:
didn't you say you did not need to use `var`?

Reply:
To clarify: `dynamic` _can_ (but it's not a requirement) be used to tell the compiler that the exact type won't be known until runtime. It's not a silver bullet, because it disable compile-type checks. E.g. if I write this:
```c#
dynamic foo = account.ExternalAccounts.Data[0];
Console.WriteLine(foo.AccountHolderName)
```

The compiler has no way of knowing whether `AccountHolderName` is a valid call. If it's not (because `foo` is a `Card` instead of a `BankAccount`), a runtime exception will be raised.

A safer but more verbose alternative:
```c#
var foo = account.ExternalAccounts.Data[0]; // foo is an `IExternalAccount`
if (foo is BankAccount)
{
  Console.WriteLine(((BankAccount)foo).AccountHolderName); // need to manually cast to BankAccount
}
else if (foo is Card)
{
  Console.WriteLine(((Card)foo).Name); // need to manually cast to Card
}
else
{
  // this should never be reached, unless a new concrete type is introduced for IExternalAccount
}
```

In the specific place of the code, we can use `var` because we don't need to access any of specific properties of the concrete types. We could replace `var` with `IExternalAccount` an it would be a no-op (it's the type that is inferred by the compiler).

Reply:
Ack thanks for the thorough explanation. So the logic becomes similar to what we do in stripe-java. More verbose but safer and cleaner.

Reply:
Sorry for the lurk, but +1 the choice to avoid `dynamic` if possible. It's one of these tools that can be useful in a huge variety of situations, but probably should be avoided in almost all of them because of loss of typedness is a very significant drawback (no more compiler checks, Intellisense, etc.).